torch/_functorch/aot_autograd.py

```diff
@@ -3164,7 +3180,8 @@ def runtime_wrapper(*args):
                     t._dynamo_weak_dynamic_indices |= o.dynamic_dims
                 else:
                     t._dynamo_weak_dynamic_indices = o.dynamic_dims.copy()
-
+        if runtime_metadata.set_grad_enabled is not None:
```

Comment:
Do we really need this? Can we just always mutate the grad_enabled state?

I believe the answer is _yes_:
1. We guard on the ambient grad mode
2. Hence, regardless of whether grad mode is mutated in graph, the grad mode at the time of compilation should be the grad mode every time we finish running the compiled graph

Still, perhaps we should rely on TracingContext for hygeine's sake? :thinking:

Reply:
I stuck with emitting lazily as per info in TracingContext